tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java

```diff
@@ -172,42 +172,43 @@ public void notifyNewBestBlock(StoredBlock block) throws VerificationException {
     }
 
     private static void writeBinaryCheckpoints(TreeMap<Integer, StoredBlock> checkpoints, File file) throws Exception {
-        final FileOutputStream fileOutputStream = new FileOutputStream(file, false);
-        MessageDigest digest = Sha256Hash.newDigest();
-        final DigestOutputStream digestOutputStream = new DigestOutputStream(fileOutputStream, digest);
-        digestOutputStream.on(false);
-        final DataOutputStream dataOutputStream = new DataOutputStream(digestOutputStream);
-        dataOutputStream.writeBytes("CHECKPOINTS 1");
-        dataOutputStream.writeInt(0);  // Number of signatures to read. Do this later.
-        digestOutputStream.on(true);
-        dataOutputStream.writeInt(checkpoints.size());
-        ByteBuffer buffer = ByteBuffer.allocate(StoredBlock.COMPACT_SERIALIZED_SIZE);
-        for (StoredBlock block : checkpoints.values()) {
-            block.serializeCompact(buffer);
-            dataOutputStream.write(buffer.array());
-            buffer.position(0);
+        try (FileOutputStream fileOutputStream = new FileOutputStream(file, false)) {
+            MessageDigest digest = Sha256Hash.newDigest();
```

Comment:
I think if you move this line up above the try block, the two try blocks could be merged into one. Or am I overlooking something?

Reply:
try-with-resources blocks can only use final variables that imlpement AutoCloseable interface, and the lines in second try are using: "MessageDigest digest = Sha256Hash.newDigest();" which doesn't meet the requirements.

Reply:
You're right, I've moved them in a single try. Don't know why I didn't merge them in the first place. Thanks!